MarkdownConverter/Converter/Needle.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MarkdownConverter.Converter
+{
+    internal struct Needle
```

Comment:
Maybe this happens later, but is this an opportunity for a `readonly struct`?

Reply:
Yup, will definitely do that in the follow-up PR with the tuple-based constructor.